applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h

```diff
@@ -109,9 +134,719 @@ namespace MPMSearchElementUtility
             }
 
         }
+        #pragma omp critical
         rGeom.SetValue(GEOMETRY_NEIGHBOURS, geometry_neighbours);
     }
 
+    inline void CreateBoundingBoxPoints(std::vector<array_1d<double, 3>>& rPointVector,
```

Comment:
Do you think you can wrap all those functions in a separate PQMPM_search_utilities class. This would increase readability and would allow to focus on the essentials for standard FEN.

Reply:
Good idea, I'll add it to the particle mechanics to do list 🚀